### ISSUE TITLE
X-PAYPAL-AUTHORIZATION not being used at all

### ISSUE THREAD

**Issue by choxnox** [angelleye/paypal-php-library#5]
I have a use-case where merchant needs to give me certain permissions for me to call APIs on their behalf. `RequestPermissions` and `GetAccessToken` APIs are working fine, and I can get access token + secret, but there is no way for me to actually utilize this access token within this library. 

For example, I can't get merchants personal data (using `GetBasicPersonalData` and `GetAdvancedPersonalData`) even though I have valid permissions because `X-PAYPAL-AUTHORIZATION` header is not being used at all.

Basically what's needed to be added to the library is function to generate value for `X-PAYPAL-AUTHORIZATION`, and then to simply adjust `CURLRequest()` functions in both `PayPal` and `PayPal_Adaptive` classes.

**Comment by keyurptl** [comment]
@angelleye 
As per code in [http://pastebin.com/TDN8mgxA](http://pastebin.com/TDN8mgxA)
`X-PAYPAL-AUTHORIZATION` mean to say just generate auth header which will be sent.
m I right ?

**Comment by keyurptl** [comment]
I have setup the same way  http://pastebin.com/TDN8mgxA in the pastebin.
But I am getting `Authentication failed. API credentials are incorrect` Error.
@angelleye @choxnox

**Comment by iMansoorAliKhan** [comment]
https://angelleye.atlassian.net/browse/PPCL-122
